processor/sumologicprocessor/aggregate_attributes_processor.go

```diff
@@ -0,0 +1,171 @@
+		pair := &aggregation{
+			attribute: config[i].Attribute,
+			prefixes:  config[i].Patterns,
+		}
+		aggregations = append(aggregations, pair)
+	}
+
+	return &aggregateAttributesProcessor{aggregations: aggregations}
+}
+
+func (proc *aggregateAttributesProcessor) processLogs(logs plog.Logs) error {
+	for i := 0; i < logs.ResourceLogs().Len(); i++ {
+		resourceLogs := logs.ResourceLogs().At(i)
+		err := proc.processAttributes(resourceLogs.Resource().Attributes())
+		if err != nil {
+			return err
+		}
+
+		for j := 0; j < resourceLogs.ScopeLogs().Len(); j++ {
+			scopeLogs := resourceLogs.ScopeLogs().At(j)
+			for k := 0; k < scopeLogs.LogRecords().Len(); k++ {
+				err := proc.processAttributes(scopeLogs.LogRecords().At(k).Attributes())
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func (proc *aggregateAttributesProcessor) processMetrics(metrics pmetric.Metrics) error {
+	for i := 0; i < metrics.ResourceMetrics().Len(); i++ {
+		resourceMetrics := metrics.ResourceMetrics().At(i)
+		err := proc.processAttributes(resourceMetrics.Resource().Attributes())
+		if err != nil {
+			return err
+		}
+
+		for j := 0; j < resourceMetrics.ScopeMetrics().Len(); j++ {
+			scopeMetrics := resourceMetrics.ScopeMetrics().At(j)
+			for k := 0; k < scopeMetrics.Metrics().Len(); k++ {
+				err := processMetricLevelAttributes(proc, scopeMetrics.Metrics().At(k))
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func (proc *aggregateAttributesProcessor) processTraces(traces ptrace.Traces) error {
+	for i := 0; i < traces.ResourceSpans().Len(); i++ {
+		resourceSpans := traces.ResourceSpans().At(i)
+		err := proc.processAttributes(resourceSpans.Resource().Attributes())
+		if err != nil {
+			return err
+		}
+
+		for j := 0; j < resourceSpans.ScopeSpans().Len(); j++ {
+			scopeSpans := resourceSpans.ScopeSpans().At(j)
+			for k := 0; k < scopeSpans.Spans().Len(); k++ {
+				err := proc.processAttributes(scopeSpans.Spans().At(k).Attributes())
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func (proc *aggregateAttributesProcessor) isEnabled() bool {
+	return len(proc.aggregations) != 0
+}
+
+func (*aggregateAttributesProcessor) ConfigPropertyName() string {
+	return "aggregate_attributes"
+}
+
+func (proc *aggregateAttributesProcessor) processAttributes(attributes pcommon.Map) error {
+	for i := 0; i < len(proc.aggregations); i++ {
+		curr := proc.aggregations[i]
+		names := []string{}
+		attrs := []pcommon.Value{}
+
+		for j := 0; j < len(curr.prefixes); j++ {
+			prefix := curr.prefixes[j]
+			// Create a new map. Unused keys will be added here,
+			// so we can check them against other prefixes.
+			newMap := pcommon.NewMap()
+			newMap.EnsureCapacity(attributes.Len())
+
+			attributes.Range(func(key string, value pcommon.Value) bool {
+				ok, trimmedKey := getNewKey(key, prefix)
+				if ok {
+					// TODO: Potential name conflict to resolve, eg.:
```

Comment:
Having an issue attached to this TODO would be nice.

Reply:
I added validation to config

Reply:
Just to confirm, does the config validation you added solve the issue mentioned in this comment? My understanding of the comment is that trimming prefixes from attribute keys can result in conflicting keys. For example `one_foo` and `two_foo` both have the same suffix `foo`, so trimming the prefixes from these keys can produce the same key. From what I can tell, the config validation only verifies that the user doesn't try to specify `pref` and `pref1` both as prefixes.

Reply:
Thats true. I created an issue in Sumo Logic repository, which I will move to this one, after the PR will be merged